src/middlewares/parameters.ts

```diff
@@ -37,11 +45,25 @@ export const isUserId = (request: UserRequest, response: Response, next: NextFun
     if (decoded.userId.toString() === request.user.id) {
       return next();
     }
+    // Question : Je retourne les bonnes erreurs, là ?
     return unauthorized(response);
   }
   return unauthenticated(response);
 };
 
-// export const isInTeamId = (request: Request, response: Response, next: NextFunction): void => {
-//   return next();
-// };
+// Checks the user is the captain of the team. If not, it will return an error
+export const isInTeamId = async (request: UserRequest, response: Response, next: NextFunction): Promise<void> => {
+  const token = getToken(request);
+  if (token) {
+    const decoded = jwt.verify(token, jwtSecret()) as Token;
+    const { userId } = decoded;
+    const user = await fetchUser(userId.toString());
```

Comment:
Il faudrait vérifier que la fonction renvoie bien un utilisateur avant de passer à la suite

Reply:
```suggestion
    const user = await fetchUser(userId);
    if (typeof user === User) {
        const { teamId } = user;
        if (request.user.teamId === user.teamId) {
          return next();
        }
    }
    return unauthorized(response);
```
sauf que `typeof user === User` ne fonctionne pas et j'arrive pas à trouver la bonne syntaxe en TypeScript
(`User` importé de `types`)